doc/faq/troubleshooting_faq.rst

```diff
@@ -114,11 +114,39 @@ provide the following information in your e-mail to the `mailing list
   the error will help you find a bug in *your* code that is causing the
   problem.
 
-* You can get very helpful debugging output from matlotlib by running your
-  script with a ``verbose-helpful`` or ``--verbose-debug`` flags and posting
-  the verbose output the lists::
+* You can get helpful debugging output from matlotlib by using the logging_
+  library in your code and posting the verbose output the lists.  A simple
+  example is given in `examples/simple_plot_debug.py`::
 
-     python simple_plot.py --verbose-helpful > output.txt
+     python simple_plot_debug.py > output.txt
+
+  If you want to put the debugging hooks in your own code, then the
+  most simple way to do so is to insert the following *before* any calls
+  to ``import matplotlib``::
+
+    import logging
+    Format = '%(name)s:%(lineno)5d - %(levelname)s - %(message)s'
+    logging.basicConfig(level=logging.DEBUG, format=Format)
+
+    import matplotlib.pyplot as plt
+
+  (the ``Format`` just gives better looking output and is optional).
```

Comment:
For the purpose of troubleshooting, I would favor the much less complicated call without setting an explicit format (it's good enough for a bug report), and also not require to run a specific file.  Something like
```
python -c "from logging import *; basicConfig(level=DEBUG); from pylab import *; plot(); show()"
```
which should be good enough to debug many "startup" problems.

For debugging a script, I would likewise document `import logging; logging.basicConfig(level=logging.DEBUG)`.